### PR TITLE
8287103: java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java fails with Xcomp

### PR DESCRIPTION
**Pull request by lmesnik**
Sync improved in test

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed (1 review required, with at least 1 reviewer)

### Issue
 * [JDK-8287103](https://bugs.openjdk.java.net/browse/JDK-8287103): java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java fails with Xcomp

### Reviewers
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)
 * [Kevin Walls](https://openjdk.java.net/census#kevinw) (@kevinjwalls - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8821/head:pull/8821` \
`$ git checkout pull/8821`

Update a local copy of the PR: \
`$ git checkout pull/8821` \
`$ git pull https://git.openjdk.java.net/jdk pull/8821/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8821`

View PR using the GUI difftool: \
`$ git pr show -t 8821`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8821.diff">https://git.openjdk.java.net/jdk/pull/8821.diff</a>

</details>

### CODE REVIEW

**Comment by AlanBateman** [comment]
Can you add `@bug 8284161 8287103` to the test description?

File: test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java [hunk 1]
```diff
@@ -40,6 +40,7 @@ public class VirtualThreadDeadlocks {
     private static final Object LOCK1 = new Object();
     private static final Object LOCK2 = new Object();
 
+    private static volatile boolean lock2Obtained = false;
```
Comment (AlanBateman) [inline test/jdk/java/lang/management/ThreadMXBean/VirtualThreadDeadlocks.java:43]: The term used in the APIs is "lock held"  so might be better to rename this to lock2Held. Can you drop the explicit init to false and also put a blank line after the declaration as it looks strange to have it run into the comment on the main method.

**Review by AlanBateman — APPROVED** [review]

**Review by kevinjwalls — APPROVED** [review]
Looks good!

**Pull request closed without merging**